### PR TITLE
webUI tests for user-group share expiration

### PR DESCRIPTION
**Pull request by individual-it**
## Description
tests to check the behaviour of the expiration dates 

more corner-cases to be done in API tests

## Related Issue

- tests for  owncloud/enterprise#2001

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Database schema changes (next release will require increase of minor version instead of patch)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link> 
- [ ] Changelog item, see [TEMPLATE](https://github.com/owncloud/core/blob/master/changelog/TEMPLATE)

### CODE REVIEW

File: tests/acceptance/features/webUISharingInternalGroups/shareWithGroupUsingExpirationDate.feature [hunk 1]
```diff
@@ -0,0 +1,257 @@
...
+    When the user shares file "lorem.txt" with group "grp1" using the webUI without closing the share dialog
+    Then the expiration date input field should not be visible for the group "grp1" in the share dialog
+
+  Scenario: expiration date is enabled for sharing with groups, user shares a file with a group
+    Given user "user1" has uploaded file "filesForUpload/lorem.txt" to "/lorem.txt"
+    And parameter "shareapi_default_expire_date_group_share" of app "core" has been set to "yes"
+    And user "user1" has logged in using the webUI
+    When the user shares file "lorem.txt" with group "grp1" using the webUI without closing the share dialog
+    Then the expiration date input field should be visible for the group "grp1" in the share dialog
+    Then the expiration date input field should be "+7 days" for the group "grp1" in the share dialog
```
Comment (phil-davis) [inline tests/acceptance/features/webUISharingInternalGroups/shareWithGroupUsingExpirationDate.feature]:
```suggestion
    And the expiration date input field should be "+7 days" for the group "grp1" in the share dialog
```

File: tests/acceptance/features/webUISharingInternalGroups/shareWithGroupUsingExpirationDate.feature [hunk 2]
```diff
@@ -0,0 +1,257 @@
...
+  Scenario: expiration date is enforced for group, user shares a file
+    Given user "user1" has uploaded file "filesForUpload/lorem.txt" to "/lorem.txt"
+    And parameter "shareapi_default_expire_date_group_share" of app "core" has been set to "yes"
+    And parameter "shareapi_enforce_expire_date_group_share" of app "core" has been set to "yes"
+    And parameter "shareapi_expire_after_n_days_group_share" of app "core" has been set to "3"
+    And user "user1" has logged in using the webUI
+    When the user shares file "lorem.txt" with group "grp1" using the webUI without closing the share dialog
+    Then the expiration date input field should be visible for the group "grp1" in the share dialog
+    And the expiration date input field should be "+3 days" for the group "grp1" in the share dialog
+    Then the information of the last share of user "user1" should include
```
Comment (phil-davis) [inline tests/acceptance/features/webUISharingInternalGroups/shareWithGroupUsingExpirationDate.feature]:
```suggestion
    And the information of the last share of user "user1" should include
```

File: tests/acceptance/features/webUISharingInternalGroups/shareWithGroupUsingExpirationDate.feature [hunk 3]
```diff
@@ -0,0 +1,257 @@
...
+  Scenario: expiration date is enforced for group, user reshares received file with the group
+    Given user "user3" has shared file "lorem.txt" with user "user1"
+    And parameter "shareapi_default_expire_date_group_share" of app "core" has been set to "yes"
+    And parameter "shareapi_enforce_expire_date_group_share" of app "core" has been set to "yes"
+    And parameter "shareapi_expire_after_n_days_group_share" of app "core" has been set to "3"
+    And user "user1" has logged in using the webUI
+    When the user shares file "lorem.txt" with group "grp1" using the webUI without closing the share dialog
+    Then the expiration date input field should be visible for the group "grp1" in the share dialog
+    And the expiration date input field should be "+3 days" for the group "grp1" in the share dialog
+    Then the information of the last share of user "user1" should include
```
Comment (phil-davis) [inline tests/acceptance/features/webUISharingInternalGroups/shareWithGroupUsingExpirationDate.feature]:
```suggestion
    And the information of the last share of user "user1" should include
```

File: tests/acceptance/features/webUISharingInternalGroups/shareWithGroupUsingExpirationDate.feature [hunk 4]
```diff
@@ -0,0 +1,257 @@
...
+
+  Scenario: expiration date is enabled but not enforced for group, user reshares received file with group
+    Given user "user3" has shared file "lorem.txt" with user "user1"
+    And parameter "shareapi_default_expire_date_group_share" of app "core" has been set to "yes"
+    And parameter "shareapi_enforce_expire_date_group_share" of app "core" has been set to "no"
+    And user "user1" has logged in using the webUI
+    When the user shares file "lorem.txt" with group "grp1" using the webUI without closing the share dialog
+    Then the expiration date input field should be visible for the group "grp1" in the share dialog
+    And the expiration date input field should be "+7 days" for the group "grp1" in the share dialog
+    Then the information of the last share of user "user1" should include
```
Comment (phil-davis) [inline tests/acceptance/features/webUISharingInternalGroups/shareWithGroupUsingExpirationDate.feature]:
```suggestion
    And the information of the last share of user "user1" should include
```

**Review by phil-davis — APPROVED** [review]
Looks good. There are only a few small comments on Then/And etc. So we can merge this and tidy up later.

**Comment by phil-davis** [comment]
Minor comments have been fixed in the target PR
